src/combinatorics/simple_graph/subgraph.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2021 Hunter Monroe. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Hunter Monroe, Kyle Miller, Alena Gusakov
+-/
+
+import combinatorics.simple_graph.basic
+
+/-!
+# Subgraphs of a simple graph
+
+This file formalizes a subgraph of a simple graph as a structure and proves a few simple lemmas.
+-/
+
+universe u
+variables {V : Type u} (G : simple_graph V)
+open simple_graph
+
+/-- A subgraph of a graph is a subset of vertices and a subset of edges such that each endpoint of
+an edge is contained in the vertex set. Subgraphs implement the `simple_graph` class. -/
+@[ext]
+structure subgraph :=
+(V' : set V)
+(adj' : V → V → Prop)
+(adj_sub : ∀ ⦃v w : V⦄, adj' v w → G.adj v w)
+(edge_vert : ∀ ⦃v w : V⦄, adj' v w → v ∈ V')
+(sym' : symmetric adj')
```

Comment:
Is it possible to use `set_like` to facilitate the construction of the subobject, like is done for other subobjects?

Reply:
@pechersky Subgraphs aren't identified with vertex sets, so unless I don't understand `set_like` (I'm not too familiar with it yet), or it won't work here.

I'm tempted to change this definition to not use primes, and to use a different name for the vertex set.  Maybe `verts`?
```lean
@[ext]
structure subgraph :=
(verts : set V)
(adj : V → V → Prop)
(adj_sub : ∀ ⦃v w : V⦄, adj v w → G.adj v w)
(edge_vert : ∀ ⦃v w : V⦄, adj v w → v ∈ verts)
(sym : symmetric adj)
```
I had previously found that it's hard to look at too many `V` vs `V'` for too long.

Reply:
OK I will use verts